src/main/scala/com/yotpo/metorikku/Metorikku.scala

```diff
@@ -9,20 +9,38 @@ import org.apache.log4j.LogManager
 object Metorikku extends App {
   val log = LogManager.getLogger(this.getClass)
   log.info("Starting Metorikku - Parsing configuration")
-  val session = Job(ConfigurationParser.parse(args))
 
-  session.config.periodic match {
-    case Some(periodic) => {
-      executePeriodicTask(periodic)
+  val configurations = ConfigurationParser.parse(args)
+
+  val jobs = configurations.map(config =>
+    new Runnable {
```

Comment:
can u try use parallel  collections instead of Runnables? its more scala like 
https://docs.scala-lang.org/overviews/parallel-collections/overview.html

Reply:
Not sure how these are related...